h2integrate/converters/grid/grid.py

```diff
@@ -0,0 +1,274 @@
+            if isinstance(buy_price, (list, np.ndarray)):
+                if len(buy_price) != n_timesteps:
+                    raise ValueError(
+                        f"electricity_buy_price length ({len(buy_price)}) "
+                        f"must match n_timesteps ({n_timesteps})"
+                    )
+                buy_price_shape = n_timesteps
+
+            else:
+                buy_price_shape = 1
+
+            self.add_input(
+                "electricity_buy_price",
+                val=self.config.electricity_buy_price,
+                shape=buy_price_shape,
+                units="USD/(kW*h)",
+                desc="Price to buy electricity from grid",
+            )
+
+        # Electricity flowing INTO grid (selling to grid)
+        self.add_input(
+            "electricity_sold",
+            val=0.0,
+            shape=n_timesteps,
+            units="kW",
+            desc="Electricity flowing into grid (selling to grid)",
+        )
+
+        # Add sell price input if configured
+        if self.config.electricity_sell_price is not None:
+            sell_price = self.config.electricity_sell_price
+            if isinstance(sell_price, (list, np.ndarray)):
+                if len(sell_price) != n_timesteps:
+                    raise ValueError(
+                        f"electricity_sell_price length ({len(sell_price)}) "
+                        f"must match n_timesteps ({n_timesteps})"
+                    )
+                sell_price_shape = n_timesteps
+            else:
+                sell_price_shape = 1
+
+            self.add_input(
+                "electricity_sell_price",
+                val=self.config.electricity_sell_price,
+                shape=sell_price_shape,
+                units="USD/(kW*h)",
+                desc="Price to sell electricity to grid",
+            )
+
+    def compute(self, inputs, outputs, discrete_inputs, discrete_outputs):
+        interconnection_size = inputs["interconnection_size"]
+
+        # Capital costs based on interconnection size
+        capex_per_kw = self.config.interconnection_capex_per_kw
+        fixed_cost = self.config.fixed_interconnection_cost
+        outputs["CapEx"] = (interconnection_size * capex_per_kw) + fixed_cost
+
+        # Fixed operating costs based on interconnection size
+        opex_per_kw = self.config.interconnection_opex_per_kw
+        outputs["OpEx"] = interconnection_size * opex_per_kw
+
+        # Variable operating costs (positive cost for buying, negative for selling)
+        varopex = 0.0
+
+        # Add buying costs if buy price is configured
+        # electricity_out represents power flowing OUT of grid (buying)
+        if self.config.electricity_buy_price is not None:
+            electricity_out = inputs["electricity_out"]
+            buy_price = inputs["electricity_buy_price"]
+            # Buying costs money (positive VarOpEx)
+            varopex += np.sum(electricity_out * buy_price)
```

Comment:
`electricity_out` is in units of (kW), and `buy_price` is in units of (USD/(kW*h)). We need to have the time step represented to make the units work out. I think time step can be hard coded for now but including it will prepare us for allowing different time steps later as well. See also #204.